src/sdltiles.cpp

```diff
@@ -732,43 +738,47 @@ void Font::draw_ascii_lines(unsigned char line_id, int drawx, int drawy, int FG)
     }
 }
 
-void invalidate_framebuffer(int x, int y, int width, int height)
+void invalidate_oversized_framebuffer( int x, int y, int width, int height )
 {
-    for (int j = 0, fby = y; j < height; j++, fby++) {
-        std::fill_n(framebuffer[fby].chars.begin() + x, width, cursecell(""));
+    for( int j = 0, fby = y; j < height; j++, fby++ ) {
+        std::fill_n( oversized_framebuffer[fby].chars.begin() + x, width, cursecell( "" ) );
     }
 }
 
-void invalidate_map_framebuffer()
+void invalidate_terminal_framebuffer()
 {
-    if( !g->w_terrain ) {
-        return;
-    }
-    for( int j = 0, fby = g->w_terrain->y; j < TERRAIN_WINDOW_HEIGHT; j++, fby++ ) {
-        std::fill_n( framebuffer[fby].chars.begin() + g->w_terrain->x, TERRAIN_WINDOW_WIDTH,
+    for( unsigned int i = 0; i < terminal_framebuffer.size(); i++ ) {
+        std::fill_n( terminal_framebuffer[i].chars.begin(), terminal_framebuffer[i].chars.size(),
                      cursecell( "" ) );
     }
 }
 
-void invalidate_overmap_framebuffer()
+void invalidate_oversized_framebuffer()
 {
-    if( !g->w_overmap ) {
-        return;
-    }
-    for( int j = 0, fby = g->w_overmap->y; j < OVERMAP_WINDOW_HEIGHT; j++, fby++ ) {
-        std::fill_n( framebuffer[fby].chars.begin() + g->w_overmap->x, OVERMAP_WINDOW_WIDTH,
+    for( unsigned int i = 0; i < oversized_framebuffer.size(); i++ ) {
+        std::fill_n( oversized_framebuffer[i].chars.begin(), oversized_framebuffer[i].chars.size(),
```

Comment:
Looks like this is the exact same logic as in `invalidate_terminal_framebuffer`. Why are those two functions instead of one with suitable parameters?